Update Node and npm to a more recent version

## TL;DR;
- Node and Npm version should be updated to a more recent one, as we are mostly developing with newer node and npm versions, also because committed lockfiles are version 2.
Lockfile version 2 is available starting at npm v7, However I think we could update it even to a more recent version.
For example, Node v16, that includes npm v8). 
- `npm install` command [here](https://github.com/asyncapi/.github/blob/24a01f622f206047c23108a6d234c01199efb61d/.github/workflows/if-nodejs-release.yml#L69) should be replaced by `npm ci` so we avoid side effects (no modification of package-lock.json).

Any strong reason to not do this?

## Long version if you are not very convinced

The following is a real use case of a bug in the produced image. It took me like an hour to figure out the "bug" (thanks to @magicmatatjahu for pairing!).

The very first Docker image of [server-api](https://hub.docker.com/layers/asyncapi/server-api/0.1.0/images/sha256-36321bde2f5068bcaed3226b46408b3ff0ce814c791c56aa850626c1cffa9255?context=explore) was built and pushed to Docker Hub by our GH workflow action [if-nodejs-release.yml](https://github.com/asyncapi/.github/blob/master/.github/workflows/if-nodejs-release.yml).

Unfortunately, after trying to deploy it to a K8s cluster, we found out that the container was failing due to a missing node module: `Error: Cannot find module '@babel/core'`. By pulling the same image locally, we discarded the problem was in K8s but in the image itself.
The surprise is that, whenever that image is built locally (e.g. in my computer), the container works like a charm.

But why?

After diving (literally, using [dive](https://github.com/wagoodman/dive) tool) into both Docker images (the one created by CI and the one created by myself locally), I noticed there was a big difference on a particular Docker layer. In particular, in the one created by `COPY package* ./`.

![Diff images using dive](https://user-images.githubusercontent.com/1083296/149553068-0956db52-85ab-4a3c-90e4-fe33ac0e0592.png)

As you can see, the `package-lock.json` filesize differs. The one created by CI is much smaller (554 kB VS 1.2 MB).

After reading both files, I noticed that the `package-lock.json` from the image created by CI was a version 1 (`"lockfileVersion": 1`), however both the one created locally and the one available in the repository source code are version 2 (`"lockfileVersion": 2`).

But how can be possible if dependencies are installed inside docker? See [Dockerfile](https://github.com/asyncapi/server-api/blob/master/Dockerfile#L16). Could be a bug with Docker? The answer is no. 
Dependencies were not built at the time `RUN npm ci` was executed, by way before. In particular, in the GH workflow action that precedes this. Inside Docker image we only run `npm ci`, which is intended for just using the `pacakge-lock.json` as the list of dependencies to install without modification.

[This line](https://github.com/asyncapi/.github/blob/24a01f622f206047c23108a6d234c01199efb61d/.github/workflows/if-nodejs-release.yml#L69) is running a `npm install` command, so all dependencies are installed. Is the GH action running an old `npm` version that doesn't support lockfile version 2? Well, considering that the image where we run is `ubuntu-latest`, the `npm` version should be `8.1.2` (according to [this](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md#package-management)). But we are not using that NPM (nor node) executable, but rather the one installed in the [previous step](https://github.com/asyncapi/.github/blob/24a01f622f206047c23108a6d234c01199efb61d/.github/workflows/if-nodejs-release.yml#L62-L66), which is:

- Node v14.18.3
- NPM v6.14.15

![Setup Node.js versions screenshot from action](https://user-images.githubusercontent.com/1083296/149557791-6f8f2d3e-2e0b-443e-9c56-92f31704c8ec.png).

The reason why the npm lockfile version is 1 in CI is because `npm install` that ran in the workflow with NPM v6.14.15 doesn't understand lockfile version 2 fully, so it literally tries to do the best (which seems it's not the best for us). As the warning says:

```bash
❯ npm install
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
```

Due to that missing dependency (`@babel/core`) that npm couldn't install, the app is broken.

## Related issues

- https://github.com/asyncapi/parser-js/issues/427

--- a/.github/workflows/link-check-cron.yml
+++ b/.github/workflows/link-check-cron.yml
@@ -28,5 +28,5 @@ jobs:
           status: ${{ job.status }}
           fields: repo,message,action,eventName,ref,workflow
       env:
-          SLACK_DOCS_CHANNEL: ${{ secrets.SLACK_DOCS_CHANNEL }}
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_DOCS_CHANNEL }}
       if: failure() # Only, on failure, send a message on the Slack Docs Channel (if there are broken links)

--- a/.github/workflows/link-check-cron.yml
+++ b/.github/workflows/link-check-cron.yml
@@ -26,7 +26,7 @@ jobs:
         # More information can be found here: https://github.com/tcort/markdown-link-check#config-file-format
         # The default file is mlc_config.json, located at the root of the directory
         
-        # config-file: name-of-config-file
+        # config-file: name-of-config-file.json
 
     - name: Report workflow run status to Slack
       uses: 8398a7/action-slack@v3

--- a/.github/workflows/link-check-cron.yml
+++ b/.github/workflows/link-check-cron.yml
@@ -24,9 +24,7 @@ jobs:
         
         # A configuration file can be included, indicating the properties of the link check action
         # More information can be found here: https://github.com/tcort/markdown-link-check#config-file-format
-        # The default file is mlc_config.json, located at the root of the directory
-        
-        # config-file: name-of-config-file.json
+        # Create mlc_config.json file in the root of the directory
 
     - name: Report workflow run status to Slack
       uses: 8398a7/action-slack@v3

--- a/.github/workflows/link-check-cron.yml
+++ b/.github/workflows/link-check-cron.yml
@@ -21,6 +21,11 @@ jobs:
       with:
         use-quiet-mode: 'yes'
         use-verbose-mode: 'yes'
+        # A configuration file can be included, indicating the properties of the link check action
+        # More information can be found here: https://github.com/tcort/markdown-link-check#config-file-format
+        # The default file is mlc_config.json, located at the root of the directory
+        
+        # config: name-of-config-file
 
     - name: Report workflow run status to Slack
       uses: 8398a7/action-slack@v3

--- a/.github/workflows/link-check-cron.yml
+++ b/.github/workflows/link-check-cron.yml
@@ -21,11 +21,12 @@ jobs:
       with:
         use-quiet-mode: 'yes'
         use-verbose-mode: 'yes'
+        
         # A configuration file can be included, indicating the properties of the link check action
         # More information can be found here: https://github.com/tcort/markdown-link-check#config-file-format
         # The default file is mlc_config.json, located at the root of the directory
         
-        # config: name-of-config-file
+        # config-file: name-of-config-file
 
     - name: Report workflow run status to Slack
       uses: 8398a7/action-slack@v3

--- a/.github/workflows/link-check-pr.yml
+++ b/.github/workflows/link-check-pr.yml
@@ -17,4 +17,10 @@ jobs:
       with:
         use-quiet-mode: 'yes'
         use-verbose-mode: 'yes'
-        check-modified-files-only: 'yes' #Only modified files are checked on PRs
+        check-modified-files-only: 'yes' # Only modified files are checked on PRs
+        
+        # A configuration file can be included, indicating the properties of the link check action
+        # More information can be found here: https://github.com/tcort/markdown-link-check#config-file-format
+        # The default file is mlc_config.json, located at the root of the directory
+        
+        # config: name-of-config-file.json

--- a/.github/workflows/link-check-pr.yml
+++ b/.github/workflows/link-check-pr.yml
@@ -23,4 +23,4 @@ jobs:
         # More information can be found here: https://github.com/tcort/markdown-link-check#config-file-format
         # The default file is mlc_config.json, located at the root of the directory
         
-        # config: name-of-config-file.json
+        # config-file: name-of-config-file.json

--- a/.github/workflows/link-check-pr.yml
+++ b/.github/workflows/link-check-pr.yml
@@ -21,6 +21,4 @@ jobs:
         
         # A configuration file can be included, indicating the properties of the link check action
         # More information can be found here: https://github.com/tcort/markdown-link-check#config-file-format
-        # The default file is mlc_config.json, located at the root of the directory
-        
-        # config-file: name-of-config-file.json
+        # Create mlc_config.json file in the root of the directory

